udev_monitor.c

```diff
@@ -338,20 +305,22 @@ struct udev_monitor *udev_monitor_unref(struct udev_monitor *udev_monitor)
         return NULL;
     }
 
-    write(udev_monitor->pfd[1], "1", 1);
-    pthread_barrier_wait(&udev_monitor->barrier);
-    pthread_barrier_destroy(&udev_monitor->barrier);
+    // Write 8 bytes to tell event thread to die
+    write(udev_monitor->signal_fd, udev_monitor, 8);
+    // waiting for event thread to end before freeing udev_monitor
+    if (udev_monitor->thread) {
```

Comment:
I don't think that this check is needed.

Reply:
Sure. If thread was uninitialized we'd just get `ESRCH` and there would be nothing for us to do.